Add controllers

ReViZe96 opened the pull request:
1. main logic for users and items
2. validation of creating/updating users and items
3. error handling
4. logging

ReViZe96 committed:
java-shareit 0.1.0:
added main logic for users and items
need refactoring and check

ReViZe96 committed:
java-shareit 0.1.1:
1. validation of creating/updating users and items
2. error handling
3. refactoring of update and find methods
4. common-lang3 to pom
5. logging

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/Item.java:
```diff
@@ -0,0 +1,17 @@
+
+@Data
+@Builder
+public class Item {
```
хорошо бы положить в папку model

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,78 @@
+     * @return DTO добавленной вещи
+     */
+    @PostMapping
+    public ItemDto addItem(@RequestBody ItemDto newItem,
```
Давай для этого приложения добавим новый класс-обертку для методов контроллера ResponseEntity - он возвращает не только json, но и http-статус, что довольно удобно. Будет выглядеть как-то так:
return ResponseEntity.ok().body(itemService.addItem(..));
Для всех методов во всех контроллерах стоит его добавить

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,78 @@
 @RequestMapping("/items")
 public class ItemController {
+
+    private ItemService itemService;
```
final

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,78 @@
+
+    private ItemService itemService;
+
+    public ItemController(ItemService itemService) {
```
чтобы в контроллерах не создавать конструкторы, мы можем поставить над классом @RequiredArgsConstructor и тогда все переменные, помеченные как final, будут автоматом добавляться в неявный конструктор - таким образом будет обеспечено внедрение зависимости

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/ItemMapper.java:
```diff
@@ -0,0 +1,26 @@
+
+import ru.practicum.shareit.user.User;
+
+public class ItemMapper {
```
для этого приложения давай добавим библиотеку mapStruct - она используется на всех реальных проектах, будет очень полезно ее изучить. Все методы маппинга будут писаться в одну строчку благодаря ней, и нам это очень облегчит работу над этим приложением в ближайших спринтах

andrew-larkin reviewed (changes requested):
Привет! Хороший старт! Правки, которые я описал для item, актуальны и для user - выполнить их и работа будет принята.

ReViZe96 committed:
java-shareit 0.1.2:
1. constructors replace on @RequiredArgsConstructor and final fields
2. ResponseEntity for all controller methods returning values
3. add using of MapStruct for entity-DTO mapping

andrew-larkin reviewed (approved):
Привет! Отлично, работа принята!
